auslib/db.py

```diff
@@ -1539,18 +1539,28 @@ def _buildIDMatchesRule(self, ruleBuildID, queryBuildID):
             return True
         return string_compare(queryBuildID, ruleBuildID)
 
-    def _csvMatchesRule(self, ruleString, queryString, substring=True):
-        """Decides whether a column from a rule matches an incoming one.
-           Some columns in a rule may specify multiple values delimited by a
-           comma. Once split we do a full or substring match against the query
-           string. Because we support substring matches, there's no need
-           to support globbing as well."""
+    def _simpleBooleanMatchesSubRule(self, subRuleString, queryStringList):
+        for rule in subRuleString:
+            if rule not in queryStringList:
+                return False
+        return True
+
+    def _simpleBooleanMatchesRule(self, ruleString, queryString):
+        """Decides whether a column from a rule matches an incoming one using simplified boolean logic.
+        Only two operators are supported: '&&' (and), ',' (or). A rule like 'AMD,SSE' will match incoming
+        rules that contain either 'AMD' or 'SSE'. A rule like 'AMD&&SSE' will only match incoming rules
+        that contain both 'AMD' and 'SSE'.
+        Rules are matched to full strings, so if we have a rule that specifies 'AMD,SSE', it will NOT be
+        matched by an incoming rule that contains 'SSE3'.
+        """
         if ruleString is None:
             return True
-        for part in ruleString.split(','):
-            if substring and part in queryString:
-                return True
-            elif part == queryString:
+
+        decomposedRules = [[rule.strip() for rule in subRule.split('&&')] for subRule in ruleString.split(',')]
+        decomposedQueryString = queryString.split(',')
```

Comment:
Ah, I see. I think it's actually best to avoid doing this for two reasons:
1) There's no reason to support lists of values in the query yet (and maintaining code that serves no purpose is wasteful)
2) It's a bit "magical" to have _simpleBooleanMatchesSubRule operate different depending on the type of the "preparedQuery". It would be better to use an explicit flag for this, when we get there.